homeassistant/components/modbus/__init__.py

```diff
@@ -266,12 +267,16 @@ def setup(self):
 
     async def read_coils(self, unit, address, count):
         """Read coils."""
+        if self._client is None:
```

Comment:
Why do we need to wait to set up the client until home assistant start event?

Reply:
We do not, this part of the code is before my time, and I thought it was mandatory to wait for the start event so I simply continued along the same line.

I will change that now, thanks for the advice, slowly I learn more about how the core works, so I can start helping others.

Reply:
Usually the reason to wait until home assistant start event is to allow other integrations that we rely on to set up. This is common in eg template integrations that take their input from other integrations.

Reply:
That is a logical explanation, but also a very crude way of doing it, which I presume cannot hold in the long term:
think a an integration depending on another integration and being used in a template integration.

Having a method wait_for_integration_to_complete(<integration>) would be a more direct way, but that is very much outside the scope of this PR.

push a new commit, with your requested changes.

Reply:
Yeah, we should probably have a way to say that an integration should wait until final call before boarding so to speak, during set up but still before start event.